### PR TITLE
[WIP] NFL tutorial draft

### PR DESCRIPTION
**Pull request by zseta**
1st rough draft of the NFL dataset tutorial with planned structure,  without queries (yet).

### CODE REVIEW

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 1]
```diff
@@ -0,0 +1,188 @@
+# Win your NFL fantasy league with TimescaleDB
+
+This tutorial is a step-by-step guide on how to ingest and analyze american football data with TimescaleDB.
+
+The dataset that we're using is provided be the National Football League (NFL) and contains data about
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:5]: "The dataset we're using is provided **BY** the..."

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 2]
```diff
@@ -0,0 +1,188 @@
+# Win your NFL fantasy league with TimescaleDB
+
+This tutorial is a step-by-step guide on how to ingest and analyze american football data with TimescaleDB.
+
+The dataset that we're using is provided be the National Football League (NFL) and contains data about 
+all the passing plays of the 2018 NFL season. We're ingesting this dataset and exploring it in fun ways. You will also see how this dataset and using TimescaleDB can improve your NFL fantasy league strategy. If you aren't an NFL fan, don't worry, you will still find this tutorial helpful as it's showcasing how to get started with TimescaleDB and explore a real world dataset with SQL and Python.
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:6]: The second sentence feels a little out of place, maybe expand on things more? Or cut it??

Comment (mirandaauhl) [reply timescaledb/tutorials/nfl-fantasy-league.md:6]: "You will also see how this dataset, and using TimescaleDB can improve your..." maybe a comma here?

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+
+1. [Create tables][]
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:8]: should we add prerequisites and about the data set in the list? It feels a little confusing to have this list then have to wait to see it manifest.

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+
+* Python 3
+* TimescaleDB (see [installation options][install-timescale]) 
+* [Psql][psql-install] or any other PostgreSQL client (e.g. DBeaver)
+* Pandas and Matplotlib (`pip install pandas matplotlib`)
+
+## About the dataset
+
+The dataset we're using in this tutorial is provided by the National Football League (NFL) as part of their yearly
+Big Data Bowl. Big Data Bowl is a contest for data professionals and students to explore american football data and build interesting machine learning models.
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:22]: "THE Big Data Bowl...."?

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+3. [Analyze NFL data][]
+   
+## Prerequisites
+
+* Python 3
+* TimescaleDB (see [installation options][install-timescale]) 
+* [Psql][psql-install] or any other PostgreSQL client (e.g. DBeaver)
+* Pandas and Matplotlib (`pip install pandas matplotlib`)
+
+## About the dataset
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:19]: This section seems a little redundant, it seems we are paraphrasing from above, maybe we should cut it or try to provide new information??

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+    gameId INT,
+    playId INT,
+    playDirection VARCHAR(5),
+    route TEXT
+);
+```
+
+## Ingest data from CSV files
+
+We have 3 separate CSV files for game, player, and play tables. For the tracking table, you will need to
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:120]: "We have three separate CSV files for the game, player, and play tables." maybe?

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 7]
```diff
@@ -0,0 +1,188 @@
...
+import data from 17 CSV files (1 file for each week in the season).
+
+Let's use a Python script that utilizes psycopg2's `copy_expert` function to ingest the data:
+
+```python
+import config
+import psycopg2
+
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:128]: should we add code comments to walk users through what each step is doing??
ex: 
"# First connect to the database location"
"# Create function to define cursor and import data"

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 8]
```diff
@@ -0,0 +1,188 @@
...
+for i in range(1, 18): 
+    print("Inserting week{i}".format(str(i)))
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Analyze NFL data
+
+Now that you have all the data ingested, let's see some ideas how you can analyze the data to help you perfect
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-fantasy-league.md:164]: "let's go over some ideas on how you can analyze the data to help..."

**Review by mirandaauhl — COMMENTED** [review]
Hey Attila, here are some initial thoughts. I think things look really good! Most all of my comments are just suggestions on re-wording, readability, and engagement so feel free to take them with a grain of salt. 

Please let me know if you need any clarification, or if I can help with anything :) 

side note: I will make sure today the highlighted queries are ready to go :)

File: timescaledb/tutorials/nfl-fantasy-league.md (hunk 2 above)
Comment (zseta) [reply timescaledb/tutorials/nfl-fantasy-league.md]: https://github.com/timescale/docs/pull/187/commits/6a12d2bc2512699e3004e5cd0a3da6bf3ce7a39f

File: timescaledb/tutorials/nfl-fantasy-league.md (hunk 1 above)
Comment (zseta) [reply timescaledb/tutorials/nfl-fantasy-league.md]: https://github.com/timescale/docs/pull/187/commits/6a12d2bc2512699e3004e5cd0a3da6bf3ce7a39f

File: timescaledb/tutorials/nfl-fantasy-league.md (hunk 3 above)
Comment (zseta) [reply timescaledb/tutorials/nfl-fantasy-league.md:12]: https://github.com/timescale/docs/pull/187/commits/25196ee59ae89b060d067d45367bbbe51b4aaa90

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 9]
```diff
@@ -0,0 +1,198 @@
+# Win your NFL fantasy league with TimescaleDB
+
+This tutorial is a step-by-step guide on how to ingest and analyze american football data with TimescaleDB.
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:3]:
```suggestion
This tutorial is a step-by-step guide on how to ingest and analyze American fantasy football data with TimescaleDB.
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 10]
```diff
@@ -0,0 +1,198 @@
...
+
+The dataset that we're using is provided by the National Football League (NFL) and contains data about 
+all the passing plays of the 2018 NFL season. We're going to ingest this dataset with Python into TimescaleDB 
+and start exploring it to discover interesting things about players that could help you win your next fantasy season. 
+If you aren't an NFL fan, don't worry, you will still find this tutorial helpful as it's showcasing how to
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:8]:
```suggestion
If you aren't an NFL fan, this tutorial can still help you  
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 11]
```diff
@@ -0,0 +1,198 @@
...
+## Prerequisites
+
+* Python 3
+* TimescaleDB (see [installation options][install-timescale]) 
+* [Psql][psql-install] or any other PostgreSQL client (e.g. DBeaver)
+* Pandas and Matplotlib (`pip install pandas matplotlib`)
+
+## Download the dataset
+
+Download dataset here!
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:25]: link?

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 12]
```diff
@@ -0,0 +1,198 @@
...
+
+## Create tables
+
+You will need to create four tables:
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:29]:
```suggestion
You need to create four tables:
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 13]
```diff
@@ -0,0 +1,198 @@
...
+
+```sql
+CREATE INDEX idx_gameid ON tracking (gameid);
+CREATE INDEX idx_playerid ON tracking (player_id);
+CREATE INDEX idx_playid ON tracking (playid);
+```
+
+## Ingest data from CSV files
+
+We have three separate CSV files for game, player, and play tables. For the tracking table, you will need to
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:127]:
```suggestion
There are three separate CSV files for game, player, and play tables. For the tracking table, you need to
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 14]
```diff
@@ -0,0 +1,198 @@
...
+import data from 17 CSV files (1 file for each week in the season).
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:128]:
```suggestion
import data from 17 CSV files (one file for each week of the season).
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 15]
```diff
@@ -0,0 +1,198 @@
...
+
+Let's use a Python script that utilizes psycopg2's `copy_expert` function to ingest the data:
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:130]:
```suggestion
You can use a Python script that uses psycopg2's `copy_expert` function to ingest the data:
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 16]
```diff
@@ -0,0 +1,198 @@
...
+for i in range(1, 18): 
+    print("Inserting week{i}".format(str(i)))
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Analyze NFL data
+
+Now that you have all the data ingested, let's go over some ideas on how you can analyze the data to help you perfect
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:174]:
```suggestion
Now that you have all the data ingested, here are some ideas for analyzing the data to help you perfect
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 17]
```diff
@@ -0,0 +1,198 @@
...
+your fantasy drafting strategy and win your fantasy season.
+
+To analyze the dataset you will need to install two libraries, if you haven't already: Pandas and Matplotlib.
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:177]:
```suggestion
To analyze the dataset you need to install two libraries, if you haven't already: Pandas and Matplotlib.
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 18]
```diff
@@ -0,0 +1,198 @@
...
+
+### Install pandas and matplotlib
+```bash
+pip install pandas matplotlib
+```
+
+1. Number of yards run in game for passing plays, by player and game 
+1. Avg yards run for a player over a game
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:185]:
```suggestion
1. Average yards run for a player over a game
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 19]
```diff
@@ -0,0 +1,198 @@
...
+1. Average and median yards run per game by type of player (not taking avg of individual)
+1. Num of snap plays by player where they were on the offense
+1. Number of plays vs points scored
+1. Average yards per game for top 3 players of each position
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:189]:
```suggestion
1. Average yards per game for top three players of each position
```

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 20]
```diff
@@ -0,0 +1,198 @@
...
+To analyze the dataset you will need to install two libraries, if you haven't already: Pandas and Matplotlib.
+
+### Install pandas and matplotlib
+```bash
+pip install pandas matplotlib
+```
+
+1. Number of yards run in game for passing plays, by player and game 
+1. Avg yards run for a player over a game
+1. Average and median yards run per game by type of player (not taking avg of individual)
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-fantasy-league.md:186]:
```suggestion
1. Average and median yards run per game by type of player (not taking the average of an individual)
```

**Review by Loquacity — CHANGES_REQUESTED** [review]
Wording suggestions.

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 21]
```diff
@@ -0,0 +1,198 @@
...
+  Information about each game.
+* player
+  
+  Player information.
+* play
+  
+  Play information
+* tracking
+  
+  Tracking information from each play.
```
Comment (ryanbooz) [inline timescaledb/tutorials/nfl-fantasy-league.md:42]: Would be great to add a few key fields from each table or something to describe this data just a little bit more.

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 22]
```diff
@@ -0,0 +1,198 @@
...
+
+```sql
+CREATE TABLE game (
+    gameId INT PRIMARY KEY,
+    gameDate DATE,
+    gameTimeEastern TIME,
+    homeTeamAbbr VARCHAR(3),
+    visitorTeamAbbr VARCHAR(3),
```
Comment (ryanbooz) [inline timescaledb/tutorials/nfl-fantasy-league.md:50]: Guaranteed to always bee 3 characters? Again, in general, the Postgres world uses TEXT for almost all text unless there is a programatic needs to have the DB limit/check the size of data and reject.

File: timescaledb/tutorials/nfl-fantasy-league.md [hunk 23]
```diff
@@ -0,0 +1,198 @@
...
+    frameId INT,
+    team TEXT,
+    gameId INT,
+    playId INT,
+    playDirection VARCHAR(5),
+    route TEXT
+);
+```
+
+Add indexes to the `tracking` table speed up queries:
```
Comment (ryanbooz) [inline timescaledb/tutorials/nfl-fantasy-league.md:117]:
```suggestion
Add indexes to the `tracking` table to improve query performance:
```

File: timescaledb/tutorials/nfl-fantasy-league.md (hunk 15 above)
Comment (ryanbooz) [reply timescaledb/tutorials/nfl-fantasy-league.md:130]: So, dumb question here. Is it worth showing an example with `COPY` or psql `/copy`?

File: timescaledb/tutorials/nfl-fantasy-league.md (hunk 17 above)
Comment (ryanbooz) [reply timescaledb/tutorials/nfl-fantasy-league.md:177]: This sounds like a requirement, which it shouldn't be (generally). Feels like we should say something like, "For each example query below, we visualize the result with Matplotlib in Python. If you want to graph the data as well, you will need to import Pandas and Matplotlib."

Remember, we want to emphasize the SQL part of this, but so far I feel like I'm only reading a Python tutorial. To me, I think it needs to be clear up front that we're analyzing the data with SQL, and using Python as a way to interact with/visualize the results. IMO

**Review by ryanbooz — CHANGES_REQUESTED** [review]
First round. Overall great stuff - quick and simple to dive in and get started which is a win for everyone!

My main comment is that we somehow need to emphasize that we're analyzing this with SQL and not Pandas/data frames like most of the stuff on Kaggle. So, maybe we showcase that at the top and make sure it's clear in both titles and intro paragraphs.

File: timescaledb/tutorials/nfl-analytics/advanced-analysis.md [hunk 24]
```diff
@@ -0,0 +1,209 @@
+## Analyze NFL data
+
+The data has been ingested, you've run a straight-forward aggregate query, and
+then you improved the performance of your analysis with continuos aggregates.
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/advanced-analysis.md:4]: continuous is spelled wrong here

File: timescaledb/tutorials/nfl-analytics/advanced-analysis.md [hunk 25]
```diff
@@ -0,0 +1,209 @@
...
+```
+
+The image below is an example of a visualization that you could create with the 
+results from this query. The scatter plot is grouped, showing the winning team's 
+plays and scores as gold, and the losing team's plays and scores as brown. 
+
+<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/nfl_tutorial/wins_vs_plays.png" alt="Wins vs Plays"/>
+
+The y-axis, or the number of plays for one team during a single game, shows that 
+more plays do not always imply a guaranteed win. In fact, the top three teams
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/advanced-analysis.md:149]: "shows that more _passing_ plays do not..."

File: timescaledb/tutorials/nfl-analytics/advanced-analysis.md [hunk 26]
```diff
@@ -0,0 +1,209 @@
...
+<img class="main-content__illustration" src="https://s3.amazonaws.com/assets.timescale.com/docs/images/tutorials/nfl_tutorial/top_3_players.png" alt="Top Three Players by Position"/>
+
+Notice that the average yards overall for Free Safety players is higher than that 
+of Wide Receivers (this is because of how we ordered the data, noted above). 
+However, individual Wide Receivers run more yards on average per game. Also, notice 
+that Kyle Juszczyk runs far more on average than other Fullback players. 
+
+## It's only halftime!
+These example queries are just the beginning examples of the analysis you can 
+perform on any tie-series data with regular SQL and helpful features like continuous
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/advanced-analysis.md:205]: time-series not tie-series??

**Review by mirandaauhl — CHANGES_REQUESTED** [review]
These are the only things I saw! Also I don't know if this needs to be specified but we may want to include the importing of the Python libraries for the field drawing tutorial part? Just a thought :)

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 27]
```diff
@@ -0,0 +1,283 @@
...
+The data is provided in multiple CSV files, each corresponding to a table in the database
+that contains the following information:
+
+* **game**
+  * Information about each game (home team, visitor team, week of play, etc.)
+  * `game_id` is a primary key
+
+* **player**
+  * Player information (display_name, college, position, etc.)
+  * `player_id` is a primary_key.
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:16]:
```suggestion
  * `player_id` is a primary key. 
```

Or:

```suggestion
  * `player_id` is a `primary_key`. 
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 28]
```diff
@@ -0,0 +1,283 @@
...
+* **game**
+  * Information about each game (home team, visitor team, week of play, etc.)
+  * `game_id` is a primary key
+
+* **player**
+  * Player information (display_name, college, position, etc.)
+  * `player_id` is a primary_key. 
+
+* **play**
+  * Play information (game, play, quarter, down, pass result, etc). _Lots_ of good
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:19]:
```suggestion
  * Play information (game, play, quarter, down, pass result). Lots of good
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 29]
```diff
@@ -0,0 +1,283 @@
...
+* **scores**
+  * Final result of each game.
+  * This table can be joined with the tracking table using the `home_team_abb` and 
+  `visitor_team_abb` fields.
+
+* **stadium_info**
+  * Home stadium of each team and additional information like `surface`, `roof_type`, `location`.
+
+
+Create the tables with the following SQL.
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:40]:
```suggestion
Create the tables with this SQL:
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 30]
```diff
@@ -0,0 +1,283 @@
...
+for i in range(1, 18): 
+    print("Inserting week{i}".format(str(i)))
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Run your first query
+
+Now that you have all the data ingested, let's run your first aggregation query
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:210]:
```suggestion
Now that you have all the data ingested, you can run the first aggregation query 
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 31]
```diff
@@ -0,0 +1,283 @@
...
+    insert("data/week{i}.csv".format(i=i), "tracking")
+
+conn.close()
+```
+
+## Run your first query
+
+Now that you have all the data ingested, let's run your first aggregation query 
+and examine the results. For most of the example queries that follow throughout 
+the tutorial, you'll need to aggregate data from the `tracking` table, which
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:212]:
```suggestion
and examine the results. For most of the example queries in this tutorial,
you'll need to aggregate data from the `tracking` table, which 
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 32]
```diff
@@ -0,0 +1,283 @@
...
+Now that you have all the data ingested, let's run your first aggregation query 
+and examine the results. For most of the example queries that follow throughout 
+the tutorial, you'll need to aggregate data from the `tracking` table, which 
+contains multiple rows per player for each play (because the data is sampled 
+multiple times per second during each play)
+
+### Number of yards run in game for passing plays, by player and game
+
+This query sums all yards for each player in every game. You can then 
+join that on the `player` table to get player details.
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:219]:
```suggestion
join that on the `player` table to get player details:
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 33]
```diff
@@ -0,0 +1,283 @@
...
+```sql
+SELECT t.player_id, p.display_name, SUM(dis) AS yards, t.gameid 
+FROM tracking t
+LEFT JOIN player p ON t.player_id = p.player_id 
+GROUP BY t.player_id, p.display_name, t.gameid 
+ORDER BY t.gameid ASC, yards DESC;
+```
+Your data should look like this:
+
+|player_id |        display_name         |       yards        |   gameid   |
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:230]: You can remove the padding from the table (although I admit it looks prettier in raw like this!)

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 34]
```diff
@@ -0,0 +1,283 @@
...
+|-----------|-----------------------------|--------------------|------------|
+|   2495454 | Julio Jones                 | 1030.6399999999971 | 2018090600|
+|   2507763 | Mike Wallace                |  940.0099999999989 | 2018090600|
+|   2552600 | Nelson Agholor              |  908.0299999999983 | 2018090600|
+|   2540158 | Zach Ertz                   |  882.0899999999953 | 2018090600|
+|   2539653 | Robert Alford               |  881.7599999999983 | 2018090600|
+|   2555383 | Jalen Mills                 |  856.1199999999916 | 2018090600|
+
+You might have noticed, however, that this data takes a long time to query because
+we have to aggregate every row in the `tracking` table in order to get the total
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:240]:
```suggestion
we have to aggregate every row in the `tracking` table to get the total
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 35]
```diff
@@ -0,0 +1,283 @@
...
+yards of each player, in each game. That's a lot of work for PostgreSQL to do
+when it needs to scan 20 million rows. On our small test machine this query 
+often takes 25-30 seconds to run.
+
+## Faster queries with continuous aggregates
+Most of the data we were interested in are based on this aggregation of the 
+`tracking` data. We often needed to know how far a player traveled on each play
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:247]:
```suggestion
`tracking` data. We wanted to know how far a player traveled on each play
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 36]
```diff
@@ -0,0 +1,283 @@
...
+or throughout each game. Rather than asking TimescaleDB to query and aggregate
+that raw data every time, we created a [continuous aggregate][] out of this base query
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:249]:
```suggestion
that raw data every time, we created a [continuous aggregate][cagg] out of this base query
```

I've noticed some problems when we use spaces in a ref link (see also L283)

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 37]
```diff
@@ -0,0 +1,283 @@
...
+```
+
+We'll use this continuous aggregate in many of the queries that follow in the
+next section. Feel free to play with other variations of this materialized data
+as you try to answer even more questions with TimescaleDB.
+
+
+[kaggle-download]: https://www.kaggle.com/c/nfl-big-data-bowl-2021/data
+[extra-download]: https://assets.timescale.com/docs/downloads/nfl_2018.zip
+[continuous aggregate]: /how-to-guides/continuous-aggregates/
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:283]:
```suggestion
[cagg]: /how-to-guides/continuous-aggregates/
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 38]
```diff
@@ -0,0 +1,283 @@
...
+CREATE MATERIALIZED VIEW player_yards_by_game
+WITH (timescaledb.continuous) AS
+SELECT t.player_id, t.gameid, t.position
+	time_bucket(INTERVAL '1 day', t."time") AS bucket,
+	SUM(t.dis) AS yards
+FROM tracking t 
+GROUP BY t.player_id, t.gameid, t.position, bucket;
+```
+
+Once the continuous aggregate is created, modify the query to use this new
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:264]:
```suggestion
When you have created the continuous aggregate, modify the query to use the 
```

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md [hunk 39]
```diff
@@ -0,0 +1,283 @@
...
+
+```sql
+SELECT pyg.player_id, p.display_name, SUM(yards) AS yards, pyg.gameid 
+FROM player_yards_by_game pyg
+LEFT JOIN player p ON pyg.player_id = p.player_id 
+GROUP BY pyg.player_id, p.display_name, pyg.gameid 
+ORDER BY pyg.gameid ASC, yards DESC;
+```
+
+We'll use this continuous aggregate in many of the queries that follow in the
```
Comment (Loquacity) [inline timescaledb/tutorials/nfl-analytics/ingest-and-query.md:276]:
```suggestion
We'll use this continuous aggregate in most of the queries in the
```

**Review by Loquacity — CHANGES_REQUESTED** [review]

File: timescaledb/tutorials/nfl-analytics/ingest-and-query.md (hunk 33 above)
Comment (ryanbooz) [reply timescaledb/tutorials/nfl-analytics/ingest-and-query.md:230]: Yeah, this is honestly just a copy/paste from the tool. I didn't feel like taking the time to _remove_ it. 😂

**Comment by ryanbooz** [comment]
Great suggestions @mirandaauhl and @Loquacity. All suggestions committed. One last thing I need to do is remove the original page (just wanted to keep it until we were near completion.

@CoolAssPuppy has asked for one other query that we'll work on tomorrow and add. I'll try to keep the active voice in mind @Loquacity! 😉

**Comment by Loquacity** [comment]
> @CoolAssPuppy has asked for one other query that we'll work on tomorrow and add. I'll try to keep the active voice in mind @Loquacity! 😉

Active voice, second person ('you"), and present tense! 💖

**Review by CoolAssPuppy — APPROVED** [review]

**Comment by Loquacity** [comment]
You've got a pretty nasty conflict in the index file 😕

**Review by Loquacity — APPROVED** [review]

File: timescaledb/tutorials/nfl-analytics/join-with-relational.md [hunk 40]
```diff
@@ -0,0 +1,25 @@
...
+
+### The Mile-High advantage
+
+In professional football, it's a well-understood phenomenon that teams that
+visit Denver's Mile-High stadium are at a disadvantage because unlike the home team
+(the Denver Broncos), they are not accustomed to playing in high altitude.
+
+Earlier we ingested stadium data. Now we can run a query to see the performance of
+players at each position when they are playing at Mile High Stadium.
+
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/join-with-relational.md:15]:
```suggestion

Use this query to compare the average acceleration and yards run of individual players when they are performing in stadiums outside of Denver versus when they are playing in Denver. The columns `avg_acc_den` and `avg_yards_den` represent the acceleration and yard statistics while in Denver.  



You can see that generally, many players have worse acceleration and average number of yards run per game while playing in Denver. 

```

File: timescaledb/tutorials/nfl-analytics/join-with-relational.md [hunk 41]
```diff
@@ -0,0 +1,25 @@
...
+### Grass vs. turf, the eternal (football) question
+
+Players often say they "feel" faster on artifical turf. How much faster are they
+in reality?
+
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/join-with-relational.md:20]:
```suggestion

Using this query you will extract the average acceleration that a player has while using turf verses grass. The column `avg_acc_turf` represents the players average acceleration while using artificial turf, and `avg_acc_grass` represents their average acceleration while on grass. 



For many players, it appears that they are indeed faster on artificial turf. This 'feeling' of increased speed may in fact be grounded in reality. 
```

File: timescaledb/tutorials/nfl-analytics/advanced-analysis.md [hunk 42]
```diff
@@ -0,0 +1,207 @@
+## Advanced analysis using SQL and TimescaleDB hyperfunctions
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/advanced-analysis.md:1]:
```suggestion
## Advanced analysis using SQL and TimescaleDB continuous aggregates and hyperfunctions
```

**Review by mirandaauhl — COMMENTED** [review]
Here are just a few comments, again we do not need to keep any of the ones outside of the new relational page. I will be putting in a pr momentarily for the relational page queries.

File: timescaledb/tutorials/nfl-analytics/join-with-relational.md [hunk 43]
```diff
@@ -0,0 +1,96 @@
...
+You can see that generally, it appears many players may have worse acceleration and average number of yards run per game while playing in Denver. 
+
+### Grass vs. turf, the eternal (football) question
+
+Players often say they "feel" faster on artifical turf. How much faster are they
+in reality?
+
+Using this query you will extract the average acceleration that a player has while using turf verses grass. The column `avg_acc_turf` represents the players average acceleration while using artificial turf, and `avg_acc_grass` represents their average acceleration while on grass. 
+
```
Comment (CoolAssPuppy) [inline timescaledb/tutorials/nfl-analytics/join-with-relational.md:63]: I think we need to add a little exposition here that explains the two tables you're joining on. Like:

In this case, we will join the `stadium_info` and `team_data` tables on the `location` of the game. After that, we'll be able to extract the `surface` data from the `stadium_info` table.

File: timescaledb/tutorials/nfl-analytics/join-with-relational.md [hunk 44]
```diff
@@ -0,0 +1,96 @@
...
+### The Mile-High advantage
+
+In professional football, it's a well-understood phenomenon that teams that
+visit Denver's Mile-High stadium are at a disadvantage because unlike the home team
+(the Denver Broncos), they are not accustomed to playing in high altitude.
+
+Earlier we ingested stadium data. Now we can run a query to see the performance of
+players at each position when they are playing at Mile High Stadium.
+
+Use this query to compare the average acceleration and yards run of individual players when they are performing in stadiums outside of Denver versus when they are playing in Denver. The columns `avg_acc_den` and `avg_yards_den` represent the acceleration and yard statistics while in Denver.
```
Comment (CoolAssPuppy) [inline timescaledb/tutorials/nfl-analytics/join-with-relational.md:16]: We may also need a similar bit of exposition to describe the data you're joining both tables on.

File: timescaledb/tutorials/nfl-analytics/index.md [hunk 45]
```diff
@@ -0,0 +1,58 @@
+# Using TimescaleDB continuous aggregates and hyperfunctions
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/index.md:1]:
```suggestion
# Analyze data using TimescaleDB continuous aggregates and hyperfunctions 
```

File: timescaledb/tutorials/page-index/page-index.js [hunk 46]
```diff
@@ -33,6 +33,28 @@ module.exports = [
...
+          {
+            title: "Analyze data using hyperfunctions",
+            href: "nfl-analytics",
+            children: [
+              {
+                title: "Ingest and query data",
+                href: "ingest-and-query"
+              },
+              {
+                title: "Using advanced SQL and hyperfunctions",
```
Comment (mirandaauhl) [inline timescaledb/tutorials/page-index/page-index.js:45]:
```suggestion
                title: "Analyze data using SQL and hyperfunctions",
```

File: timescaledb/tutorials/nfl-analytics/index.md [hunk 47]
```diff
@@ -0,0 +1,58 @@
...
+of how to ingest time-series data into TimescaleDB (even when it doesn't _seem_ like
+time-series data), how you can use plain SQL and TimescaleDB hyperfunctions to do 
+powerful data analysis, and also visualize the data with Python.
+
+This tutorial has a few sections to help you on your journey:
+
+1. [Ingest and query data](/tutorials/nfl-analytics/ingest-and-query)
+
+    Download the data, create tables in TimescaleDB, and run your first query on NFL tracking data.
+2. [Analyze NFL data using continuous aggregates and hyperfunctions](/tutorials/nfl-analytics/advanced-analysis/)
```
Comment (mirandaauhl) [inline timescaledb/tutorials/nfl-analytics/index.md:28]:
```suggestion
2. [Analyze data using continuous aggregates and hyperfunctions](/tutorials/nfl-analytics/advanced-analysis/)
```

**Review by mirandaauhl — COMMENTED** [review]
update some of the title names to be more consistent with index

**Review by mirandaauhl — APPROVED** [review]

**Pull request merged**